lib/compat/wordpress-6.7/compat.php

```diff
@@ -40,7 +40,7 @@ function _gutenberg_add_block_templates_from_registry( $query_result, $query, $t
 	}
 
 	if ( ! isset( $query['wp_id'] ) ) {
-		$template_files = _gutenberg_get_block_templates_files( $template_type, $query );
+		$template_files = _gutenberg_get_block_templates_files( $template_type );
```

Comment:
If the issue is related to filtering templates by `post_type`, what if we try removing only the problematic `post_type` key instead of the entire `$query` array?
This might resolve the issue while maintaining relatively good performance, as it would remove the need to fetch all block template files.

What do you think, @Aljullu?

```suggestion
        // Inline comment explaining why the $query is not used.
        $template_files_query = $query;
        unset( $template_files_query['post_type'] );
        $template_files = _gutenberg_get_block_templates_files( $template_type, $template_files_query );
```

Additionally, adding an inline comment to explain why the query array isn’t passed to the `_gutenberg_get_block_templates_files()` function would make it easier to understand and reduce the risk of it being overwritten later.

Reply:
Makes sense, I applied the changes in 1641d59468188dc87a0bb3527fdfbb2873b40a9f.

Reply:
It would be helpful to add a unit test to prevent this bug from reoccurring, but it’s not a blocker.

Reply:
Good call, I added the tests directly to the backport PR in https://github.com/WordPress/wordpress-develop/pull/7676. I'm happy to add tests in Gutenberg too, but I see there are no unit tests for this part of the codebase in Gutenberg, so I assumed it's better to write them in WordPress directly? What do you think? :thinking: